src/dispatch/search/views.py

```diff
@@ -40,11 +41,6 @@ def search(
 
     # add a filter for restricted incidents
     if current_user.role != UserRoles.admin:
-        filtered_results = []
-        for r in results:
-            if r["type"].lower() == "incident":
-                if r["content"].visibility != Visibility.open:
-                    continue
-            filtered_results.append(r)
-        results = filtered_results
-    return {"query": q, "results": results}
+        results["Incident"] = [i for i in results["Incident"] if i.visibility != Visibility.open]
```

Comment:
Looks like we're returning restricted incidents instead of filtering them out here if I'm interpreting it right.